services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/IslMinPortSpeed.groovy

```diff
@@ -0,0 +1,53 @@
+package org.openkilda.functionaltests.spec.links
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.messaging.info.event.IslChangeType.DISCOVERED
+import static org.openkilda.messaging.info.event.IslChangeType.MOVED
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.model.topology.TopologyDefinition.Isl
+
+import spock.lang.Narrative
+
+@Narrative("""Minimal port speed value is chosen for ISL capacity.
+Sometimes an ISL have different port speed on its edges.
+In that case, we need to set ISL capacity and all bandwidth parameters according to minimal speed value.
+Eg. 10G on one side, and 1G on another side, the ISL should have a 1G capacity.""")
+class IslMinPortSpeed extends BaseSpecification {
+    def "System is able to set min port speed for isl capacity"() {
+        given: "Two ports with different port speed"
+        def isl = topology.islsForActiveSwitches.find { it.getAswitch()?.inPort && it.getAswitch()?.outPort }
+        def port = northbound.getPort(isl.srcSwitch.dpId, isl.srcPort)
+        assumeTrue("Wasn't able to find required a-switch links", isl as boolean)
+
+        def notConnectedIsls = topology.notConnectedIsls
+        def newDst = notConnectedIsls.find {
+            northbound.getPort(it.srcSwitch.dpId, it.srcPort).maxSpeed != port.maxSpeed
+        }
+        assumeTrue("Wasn't able to find a port with other port speed", newDst as boolean)
+        def newDstPort = northbound.getPort(newDst.srcSwitch.dpId, newDst.srcPort)
+
+        when: "Replug one end of the connected link to the destination switch(isl.srcSwitchId -> newDst.srcSwitchId)"
+        def newIsl = islUtils.replug(isl, false, newDst, true)
+
+        islUtils.waitForIslStatus([newIsl, newIsl.reversed], DISCOVERED)
+        islUtils.waitForIslStatus([isl, isl.reversed], MOVED)
+
+        then: "Max bandwidth of new ISL is equal to the minimal port speed"
+        Wrappers.wait(WAIT_OFFSET) {
+            islUtils.getIslInfo(newIsl).get().maxBandwidth == [port.maxSpeed, newDstPort.maxSpeed].min()
+        }
+
+        when: "Replug the link back where it was"
```

Comment:
```
when: "Replug the link back where it was"
then: "Cleanup: Delete MOVED ISL"
```
Such pattern is not common for us. every `when` block is considered part of the test and should have a meaningful `then` partner (not cleanup). Since `Replug the link back where it was` is not part of the test itself, I think these two steps should be merged into
```
and: "Cleanup: replug the ISL back and delete the moved one"
//...
```

Reply:
fixed